watertap/core/zero_order_unit.py

```diff
@@ -0,0 +1,264 @@
+        default=False,
+        domain=In([False]),
+        description='Holdup construction flag - must be False',
+        doc='''Zero order models do not include holdup'''))
+    CONFIG.declare('property_package', ConfigValue(
+        default=useDefault,
+        domain=is_physical_parameter_block,
+        description='Property package to use for control volume',
+        doc='''Property parameter object used to define property  calculations,
+        **default** - useDefault.
+        **Valid values:** {
+        **useDefault** - use default package from parent model or flowsheet,
+        **PhysicalParameterObject** - a PhysicalParameterBlock object.}'''))
+    CONFIG.declare('property_package_args', ConfigBlock(
+        implicit=True,
+        description='Arguments to use for constructing property packages',
+        doc='''A ConfigBlock with arguments to be passed to a property block(s)
+        and used when constructing these, **default** - None.
+        **Valid values:** {see property package for documentation.}'''))
+
+    def build(self):
+        super().build()
+
+        # Get units metadata
+        units_meta = \
+            self.config.property_package.get_metadata().get_derived_units
+
+        # Create state blocks for inlet and outlets
+        tmp_dict = dict(**self.config.property_package_args)
+        tmp_dict["has_phase_equilibrium"] = False
+        tmp_dict["defined_state"] = True
+
+        self.properties_in = self.config.property_package.build_state_block(
+            self.flowsheet().time,
+            doc="Material properties at inlet",
+            default=tmp_dict)
+
+        tmp_dict_2 = dict(**tmp_dict)
+        tmp_dict_2["defined_state"] = False
+
+        self.properties_out = self.config.property_package.build_state_block(
+            self.flowsheet().time,
+            doc="Material properties at outlet",
+            default=tmp_dict_2)
+        self.properties_waste = self.config.property_package.build_state_block(
+            self.flowsheet().time,
+            doc="Material properties at waste outlet",
+            default=tmp_dict_2)
+
+        # Create Ports
+        self.add_port("inlet", self.properties_in, doc="Inlet port")
+        self.add_port("outlet", self.properties_out, doc="Outlet port")
+        self.add_port("waste", self.properties_waste, doc="Waste port")
+
+        # Add performance variables
+        self.water_recovery = Var(
+            self.flowsheet().time,
+            initialize=0.8,
+            domain=NonNegativeReals,
+            units=pyunits.dimensionless,
+            bounds=(1E-8, 1.0000001),
+            doc='Water recovery fraction')
+        self.removal_fraction = Var(
+            self.flowsheet().time,
+            self.config.property_package.solute_set,
+            domain=NonNegativeReals,
+            initialize=0.01,
+            units=pyunits.dimensionless,
+            doc='Solute removal fraction')
+
+        self.deltaP_outlet = Var(
```

Comment:
I'm not sure that `deltaP_outlet` makes sense to have in all cases. For example, with RO, `deltaP_outlet` would be the pressure at the permeate outlet (typically atmospheric or slightly higher and on one side of the membrane if considering a flat sheet element) minus the feed pressure at the inlet. @kurbansitterley would probably have some better insight on when/where it makes sense to have both `deltaP_outlet` and `deltaP_waste` for WT3 models, but from my perspective, it seems like we would only be concerned with `deltaP_waste`. If that is true, we could eliminate `deltaP_outlet` and eventually rename `deltaP_waste` as `deltaP`.

Reply:
I am not sure that is a good idea - there could be cases where we need both (or even cases where we want `deltaP_outlet` and not `deltaP_waste`). However, there are ways we can make these terms optional, and let the user or the derived classes decide which terms to include.

Two ideas I have has so far:

1. Have `has_pressure_change` config arguments for each outlet, and rely on the user to set them. This has some minor issues for cases where a term is not optional however, as the config argument will still exist (but only have one valid value).
2. Have a private attribute on the unit model which the derived class sets before calling `super().build()`. This hides the option from the user and lets the unit model developer make the decision, which is probably a better option as the unit model would also then know whether it needs to look up a value for `deltaP` from the database.

Reply:
After reading your comments, I think it is best to keep the two options, and it looks like @andrewlee94 implemented his second option. With regard to RO, @kurbansitterley, I think these examples would be resolved by specifying the permeate outlet pressure explicitly rather than the pressure change between feed inlet and permeate outlet. Nevertheless, keeping both deltaP variables seems like the way to go.